ZFSin/zfs/cmd/zdb/zdb.c

```diff
@@ -2462,43 +2464,56 @@ dump_config(spa_t *spa)
 static void
 dump_cachefile(const char *cachefile)
 {
-	int fd;
-	struct stat statbuf;
 	char *buf;
 	nvlist_t *config;
-
-	if ((fd = open(cachefile, O_RDONLY)) < 0) {
+	int err;
+	zfs_file_t hFile;
+	zfs_file_attr_t attr;
+	ssize_t resid;
+	char pathname[MAXPATHLEN];
+
+	// unpack cachefile path
+	if (strncmp(cachefile, "\\SystemRoot\\", 12) == 0) {
+		(void)snprintf(pathname, MAXPATHLEN, "%s\\%s",
+			getenv("SystemRoot"), cachefile + 12);
+	} else
+		(void)snprintf(pathname, MAXPATHLEN, "%s", cachefile);
+
+	err = zfs_file_open(pathname, O_RDONLY, 0, &hFile);
+	if (err) {
 		(void) printf("cannot open '%s': %s\n", cachefile,
-		    strerror(errno));
+	    strerror(errno));
 		exit(1);
 	}
 
-	if (fstat(fd, &statbuf) != 0) {
+	err = zfs_file_getattr(&hFile, &attr);
+	if (err) {
 		(void) printf("failed to stat '%s': %s\n", cachefile,
-		    strerror(errno));
+	    strerror(errno));
 		exit(1);
 	}
 
-	if ((buf = malloc(statbuf.st_size)) == NULL) {
+	if ((buf = (char*)umem_alloc(attr.zfa_size, UMEM_NOFAIL)) == NULL) {
```

Comment:
If needed, should we wrap our own version of malloc() to call umem_alloc? Is umem_alloc() required here?

Reply:
Another option is to to have the whole function as separate windows version.

Reply:
Most of the userland programs use `umem_alloc` and it is a common code https://github.com/openzfsonwindows/ZFSin/blob/master/ZFSin/zfs/lib/libspl/include/umem.h#L84-L94
that's why we have used `umem_alloc`